tests/UnitTests/DurableDemographicTests/DurableDemographicTests.cs

```
@@ -176,27 +176,41 @@ public async Task RunOrchestrator_ActivityTimesOut_ReturnFalseAndLogError()
                 It.IsAny<string>(),
                 It.IsAny<TaskOptions>()
             ))
-            .Returns(neverEndingTask.Task);
+             .Returns(Task.Delay(Timeout.Infinite).ContinueWith(_ => false));
 
         mockContext
             .Setup(c => c.CreateTimer(expirationTime, It.IsAny<CancellationToken>()))
             .Returns(Task.CompletedTask);
 
-        // Act
-        bool result = await sut.RunOrchestrator(mockContext.Object);
+        mockContext.Setup(ctx => ctx.CreateReplaySafeLogger(It.IsAny<string>())).Returns(_logger.Object);
+        mockContext.Setup(ctx => ctx.GetInput<string>()).Returns("[{\"NhsNumber\": \"111111\", \"FirstName\": \"Test\"}]");
+
+        bool result = false;
+        try
+        {
+            // Act
+            result = await sut.RunOrchestrator(mockContext.Object);
+        }
+        catch
+        {
+            Assert.IsFalse(result);

```

why would we only assert when RunOrchestrator throws? this unit test is hiding an exception throw?

If the code is correct could we assert that the exception is thrown correctly?

---

that's a good thought yes. I have updated this

---

Amazing thanks!